org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java

```diff
@@ -84,8 +88,24 @@
 
 	private PreferenceManager preferenceManager;
 
+	private IPreferencesChangeListener preferencesChangeListener = new IPreferencesChangeListener() {
+
+		@Override
+		public void preferencesChange(Preferences oldPreferences, Preferences newPreferences) {
+			if (contentProviderManager == null) {
+				return;
+			}
+			List<String> oldIds = oldPreferences.getPreferredContentProviderIds();
+			List<String> newIds = newPreferences.getPreferredContentProviderIds();
+			boolean clean = Objects.equals(oldIds, newIds);
```

Comment:
you missed my correction earlier :-) boolean clean = !Objects.equals(oldIds, newIds);